### DISCUSSION TITLE
Where to start? [Metaverse Development]

### DISCUSSION THREAD

**Discussion by cruzluna** [AngellXR/metaversedev#9] [Q&A]
_TLDR_: What should I start learning to become a Metaverse developer?

Hello,  I am a college student studying computer science and I want to become a 'Metaverse Developer'. I've recently started learning Unity & C#. More specifically, I am working with the Oculus Interaction SDK and I plan on doing some projects using that library. I am not the artist type so I don't know if I want to get into Blender or Maya, for now. 

Am I on the right track with C#/Unity? What other technologies should I look into, considering I am sort of a blank slate? 

Thanks!

**Comment by mrmetaverse** [comment]
Hey! Thats a great question, and hopefully we can thread some alternatives to this, but here are some of my recommendations! Unity and C# is a solid place to start. Personally, I am big on the web-based-XR stuff. So I can also recommend checking out the following: 

- Threejs: a cross-browser JavaScript library and application programming interface used to create and display animated 3D computer graphics in a web browser using WebGL.
  - [Threejsjourney by Bruno Simon](https://threejs-journey.xyz/) is the perfect primer for threejs 
  - [Nik Lever](https://niklever.com/) has a few courses as well that I have enjoyed 
  - [Threejs documentation](https://threejs.org/) is always a great place to start as well. 
  - [react-three-fiber](https://docs.pmnd.rs/react-three-fiber/getting-started/introduction)
    - [Sougen](https://sougen.co/) is a great example of a platform in the Metaverse using r3f 
- Shaders with glsl: GLSL stands for openGL Shading Language, which is the specific standard of shader programs
  - [The book of shaders](https://thebookofshaders.com/) 
  - [ShaderToy](https://www.shadertoy.com/) is a great place to learn about shaders, and see them in action. Some of the examples are quite mind-blowing. 
- Blender: Even if you don't consider yourself artsy, it can be worth following along on a few of these as 3d modeling is intimately connected to the Metaverse. 
  - [ducky3d](https://www.youtube.com/channel/UCuNhGhbemBkdflZ1FGJ0lUQ) has loads of valuable tutorials on motion graphics, procedural materials, and geometry nodes. They are usually quick (like 15 min) and he doesn't skip steps. 
  - [Blenderguru](https://www.youtube.com/c/BlenderGuruOfficial) has been at it awhile, and makes learning modeling a lot of fun. 
  - [blender studio](https://studio.blender.org/welcome/) tutorials 
- Solidity: If you have any interest in understanding the blockchain elements of the Metaverse, I recommend exploring Solidity. Solidity is an object-oriented programming language for writing smart contracts. It is used for implementing smart contracts on various blockchain platforms, most notably, Ethereum. 
  - [Standard documentation](https://docs.soliditylang.org/en/v0.8.13/)


I am curious what others would recommend :eyes:

**Reply by jsejms** [reply]
Shaders / GLSL / OpenGL and Three.js I'd strongly recommend.  I'm more on the web XR side though.  C# and Unity is great, can't go wrong with all that.

**Comment by mrmetaverse** [comment]
I think we should have a hackathon or workshop to brainstorm our first 1-3 course-paths.

**Comment by hassping** [comment]
> _TLDR_: What should I start learning to become a Metaverse developer?
> 
> Hello, I am a college student studying computer science and I want to become a 'Metaverse Developer'. I've recently started learning Unity & C#. More specifically, I am working with the Oculus Interaction SDK and I plan on doing some projects using that library. I am not the artist type so I don't know if I want to get into Blender or Maya, for now.
> 
> Am I on the right track with C#/Unity? What other technologies should I look into, considering I am sort of a blank slate?
> 
> Thanks!



> _TLDR_: What should I start learning to become a Metaverse developer?
> 
> Hello, I am a college student studying computer science and I want to become a 'Metaverse Developer'. I've recently started learning Unity & C#. More specifically, I am working with the Oculus Interaction SDK and I plan on doing some projects using that library. I am not the artist type so I don't know if I want to get into Blender or Maya, for now.
> 
> Am I on the right track with C#/Unity? What other technologies should I look into, considering I am sort of a blank slate?
> 
> Thanks!

You must learn about IOTA at wiki.iota.org. They are developing a family of frameworks for web3 like: Decentralized Identifies, Oracles, parallel Smart Contracts, Data Streams, Tokenization, among others.

    Open source & open standard.
    Scalable
    Energy efficient
    Feeless

We are combining multimedia WebGL with IOTA to create one of the first feeless games and NFT marketplaces (zentangle.io)
